base/linalg/dense.jl

```diff
@@ -74,10 +71,8 @@ julia> isposdef(A)
 true
 ```
 """
-function isposdef(A::AbstractMatrix{T}) where T
-    S = typeof(sqrt(one(T)))
-    isposdef!(S == T ? copy(A) : convert(AbstractMatrix{S}, A))
-end
+isposdef(A::AbstractMatrix{T}) where {T} =
+    isposdef!(copy_oftype(A, promote_type(typeof(chol(one(T))),Float32)))
```

Comment:
doesn't need to be addressed in this PR, but does that indicate that maybe the other instance of this Float32 promotion should maybe get the same treatment?